File: lab_python_oop/Circle.py
import math
from .GeometricFigure import GeometricFigure
from .Color import Color

class Circle(GeometricFigure):
    name = "Круг"

    def __init__(self, radius: float, color: str):
        if radius <= 0:
            raise ValueError("Radius must be positive")
        self.radius = float(radius)
        self.color = Color(color)

    def area(self) -> float:
        return math.pi * (self.radius ** 2)

    @classmethod
    def get_name(cls) -> str:
        return cls.name

    def __repr__(self) -> str:
        return "{name} цвета: {color} радиус={radius}, площадь={area:.2f}".format(
            name=self.get_name(),
            color=self.color,
            radius=self.radius,
            area=self.area()
        )

File: geometry_project/src/oop/circle.rs
use std::f64::consts::PI;
use super::{GeometricShape, Color};

pub struct Circle {
    radius: f64,
    color: Color,
}

impl Circle {
    pub fn new(radius: f64, color: Color) -> Self {
        Self { radius, color }
    }
}

impl GeometricShape for Circle {
    fn area(&self) -> f64 {
        PI * self.radius * self.radius
    }

    fn shape_name(&self) -> &'static str {
        "Круг"
    }
}
